core/dbt/clients/jinja.py

```diff
@@ -632,3 +647,17 @@ def _convert_function(value: Any, keypath: Tuple[Union[str, int], ...]) -> Any:
     # when the test node was created in _parse_generic_test.
     kwargs = deep_map_render(_convert_function, node.test_metadata.kwargs)
     context[GENERIC_TEST_KWARGS_NAME] = kwargs
+
+
+def get_supported_languages(node: jinja2.nodes.Macro) -> List[ModelLanguage]:
+    no_args = not node.args or not node.defaults
+    dif_args_and_defaults = len(node.args) != len(node.defaults)
```

Comment:
This could be causing issue if we decided to add an arg to materialization Macro in the future. Since `kwargs` are always going to be after `args`, we can use the index to end to get things.

```
node.args = ["arg1", "supported_language", "another_key_for_kwarg"]
node.defaults = [['sql'], {"something_else": 1}]
```
you can get supported language by `node.defaults[-(len(node.args) - lang_idx)]`